src/Enchantments.cpp

```diff
@@ -167,6 +194,141 @@ bool cEnchantments::IsEmpty(void) const
+	case enchFortune:              return 3;
+	case enchPower:                return 5;
+	case enchPunch:                return 2;
+	case enchFlame:                return 1;
+	case enchInfinity:             return 1;
+	case enchLuckOfTheSea:         return 3;
+	case enchLure:                 return 3;
+	}
+	LOGWARNING("Unknown enchantment ID %d", a_EnchantmentID);
+	return 0;
+}
+
+
+
+
+
+int cEnchantments::GetMultiplier(int a_EnchantmentID, bool WithBook)
+{
+	if (WithBook)
+	{
+		switch (a_EnchantmentID)
+		{
+		case enchProtection:           return 1;
+		case enchFireProtection:       return 1;
+		case enchFeatherFalling:       return 1;
+		case enchBlastProtection:      return 2;
+		case enchProjectileProtection: return 1;
+		case enchRespiration:          return 2;
+		case enchAquaAffinity:         return 2;
+		case enchThorns:               return 4;
+		case enchDepthStrider:         return 2;
+		case enchSharpness:            return 1;
+		case enchSmite:                return 1;
+		case enchBaneOfArthropods:     return 1;
+		case enchKnockback:            return 1;
+		case enchFireAspect:           return 2;
+		case enchLooting:              return 2;
+		case enchEfficiency:           return 1;
+		case enchSilkTouch:            return 4;
+		case enchUnbreaking:           return 1;
+		case enchFortune:              return 1;
+		case enchPower:                return 1;
+		case enchPunch:                return 2;
+		case enchFlame:                return 2;
+		case enchInfinity:             return 4;
+		case enchLuckOfTheSea:         return 2;
+		case enchLure:                 return 2;
+		}
+	}
+	else  // Without book
+	{
+		switch (a_EnchantmentID)
+		{
+		case enchProtection:           return 1;
+		case enchFireProtection:       return 2;
+		case enchFeatherFalling:       return 2;
+		case enchBlastProtection:      return 4;
+		case enchProjectileProtection: return 2;
+		case enchRespiration:          return 4;
+		case enchAquaAffinity:         return 4;
+		case enchThorns:               return 8;
+		case enchDepthStrider:         return 4;
+
+		case enchSharpness:            return 1;
+		case enchSmite:                return 2;
+		case enchBaneOfArthropods:     return 2;
+		case enchKnockback:            return 2;
+		case enchFireAspect:           return 4;
+		case enchLooting:              return 4;
+
+		case enchEfficiency:           return 1;
+		case enchSilkTouch:            return 8;
+		case enchUnbreaking:           return 2;
+		case enchFortune:              return 4;
+		case enchPower:                return 1;
+		case enchPunch:                return 4;
+		case enchFlame:                return 4;
+		case enchInfinity:             return 8;
+		case enchLuckOfTheSea:         return 4;
+		case enchLure:                 return 4;
+		}
+	}
+	LOGWARNING("Unknown enchantment ID %d", a_EnchantmentID);
+	return 0;
+}
+
+
+
+
+
+bool cEnchantments::IsCompatibleWith(int a_EnchantmentID) const
+{
+	for (auto mutex: m_IncompatibleEnchantments)
```

Comment:
Calling it `mutex` is misleading us to think about thread sync. How about `excl`?